test/components/calculator-process.test.js

```diff
@@ -0,0 +1,64 @@
+const calculatorProcess = require("../../src/services/calculator-process")
+
+describe("calculator's visor test", () => {
+  const operations = [
+    ["10", "+","30","=","40"],
+    ["10", "+","10","=","20"],
+    ["20", "-","10","=","10"],
+    ["5", "*","2","=","10"],
+    ["30", "/","10","=","3"],
+    ["10", "%","1","=","0.1"],
+  ];
+
+	it.each(operations)("'s printing a basic mathematic operation", (oper1,oper2,oper3,oper4,oper5) => {
+		
+    let visorMock = jest.fn()
+    
+    calculatorProcess(oper1,visorMock)
+    calculatorProcess(oper2,visorMock)
+    calculatorProcess(oper3,visorMock)
+    calculatorProcess(oper4,visorMock)
+    calculatorProcess("C",visorMock)
+    
+    expect(visorMock.mock.calls[0][0]).toBe(oper1);
+    expect(visorMock.mock.calls[1][0]).toBe(oper1 + oper2);
+    expect(visorMock.mock.calls[2][0]).toBe(oper1 + oper2 + oper3);
+    expect(visorMock.mock.calls[3][0]).toBe(oper5);
+	});
+
+  it("'s working clear all button", () => {
+    
+    let visorMock2 = jest.fn()
+
+    calculatorProcess("C",visorMock2)
+
+    expect(visorMock2.mock.calls[0][0]).toBe("");
```

Comment:
how do you know that it's cleanup your visor, if you are not adding any number?

I guess you need to call `calculatorProcess("1",visorMock2)` and check if you get a 1 into the visor and after that call `C`

Reply:
i deleted this test, i put **"C"** in all the other operations